twake/backend/node/src/services/channels/services/member/service.ts

```diff
@@ -874,4 +878,29 @@ export class MemberServiceImpl {
       OperationType.UPDATE,
     );
   }
+
+  /**
+   * list users who have seen the message.
+   *
+   * @param {String} id - the message id
+   * @param {ChannelExecutionContext} context - the thread execution context
+   * @returns { Promise<string[]>} - the promise containing the user id list
+   */
+  async getChannelMessageSeenByUsers(
+    id: string,
+    context: ChannelExecutionContext,
+  ): Promise<string[]> {
+    const channelReadSections = await this.getChannelMembersReadSections(context);
+
```

Comment:
I propose we use the channel-stats “member_count” value to refuse executing this function if count is > 100.